### PR TITLE
feat(YouTube - Hide Suggestions shelf): Hide `Suggestions shelf`

### LINKED ISSUE #494: feat(YouTube) : add hide-suggestion-shelf patch
### Application

Youtube ReVanced

### Issue

Google added noisy reccomendation sections that i dont want to see

### Patch

Hide suggestion shelf
Hides following shelves:
- Breaking news
- Continue watching
- Explore more channels
- Shopping
- Watch it Again

### Motivation

This will save time for users not want to see this sections

### Acknowledgements

- [x] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

### PR DESCRIPTION
**Pull request by johnconner122**
closes: ReVanced/revanced-patches-template#494, ReVanced/revanced-patches-template#469

depends on:

 - https://github.com/ReVanced/revanced-integrations/pull/459

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/layout/utils/navbarindexhook/fingerprints/NavBarBuilderFingerprint.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package app.revanced.patches.youtube.layout.utils.navbarindexhook.fingerprints
+
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+import org.jf.dexlib2.AccessFlags
+import org.jf.dexlib2.Opcode
+
```
Comment (YT-Advanced) [inline src/main/kotlin/app/revanced/patches/youtube/layout/utils/navbarindexhook/fingerprints/NavBarBuilderFingerprint.kt:6]: @johnconner122 Can you help me navigate this fingerprint in JADX, I cannot found it

Comment (johnconner122) [reply src/main/kotlin/app/revanced/patches/youtube/layout/utils/navbarindexhook/fingerprints/NavBarBuilderFingerprint.kt:6]: `class public final Laeuo;`

`.super Ljava/lang/Object;`

And method name: `onClick`

YouTube v18.23.35

Comment (YT-Advanced) [reply src/main/kotlin/app/revanced/patches/youtube/layout/utils/navbarindexhook/fingerprints/NavBarBuilderFingerprint.kt:6]: Thanks

Comment (johnconner122) [reply src/main/kotlin/app/revanced/patches/youtube/layout/utils/navbarindexhook/fingerprints/NavBarBuilderFingerprint.kt:6]: That's not it, I'll let you know in a while.

Comment (YT-Advanced) [reply src/main/kotlin/app/revanced/patches/youtube/layout/utils/navbarindexhook/fingerprints/NavBarBuilderFingerprint.kt:6]: I have checked and cannot found it. Can you share the defpackage too ?

Comment (johnconner122) [reply src/main/kotlin/app/revanced/patches/youtube/layout/utils/navbarindexhook/fingerprints/NavBarBuilderFingerprint.kt:6]: ###### Class defpackage.aeuq (aeuq)
.class public abstract Laeuq;
.super Landroid/widget/HorizontalScrollView;
.source "PG"


method: `.method public m(Landroid/content/Context;)V`

Comment (YT-Advanced) [reply src/main/kotlin/app/revanced/patches/youtube/layout/utils/navbarindexhook/fingerprints/NavBarBuilderFingerprint.kt:6]: Thanks

**Comment by johnconner122** [comment]
If you are in `Library Tab` and click `search-button` or `notifications-button` and come back to `Library Tab`, NavBar index sets to zero on library tab and playlists shelf gets hidden. If anyone has any ideas to fix this, I'm all ears.

**Comment by 0xrxL** [comment]
> If you are in `Library Tab` and click `search-button` or `notifications-button` and come back to `Library Tab`, NavBar index sets to zero on library tab and playlists shelf gets hidden. If anyone has any ideas to fix this, I'm all ears.

Make sure that the litho filter check works and recognizes ```library_recent_shelf```, this is needed when you use back button and previous library tab is shown.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Pull request closed without merging**